p2p/stream/types/stream.go

```diff
@@ -41,7 +49,7 @@ type StreamID string
 
 // Meta return the StreamID of the stream
 func (st *BaseStream) ID() StreamID {
-	return StreamID(st.raw.ID())
+	return StreamID(st.raw.Conn().ID())
```

Comment:
Because previous one was wrong. I am using this ID to avoid setting up duplicate stream to a node in stream manager. But libp2p can setup multiple streams with different streamID under one connection. Thus stream.ID() cannot serve well to deduplicate streams.

For example, connection ID is 

```
Qwxxxxxxxxxxx
```

And stream IDs can be

```
Qwxxxxxxxxxxx-yyyyyyy
Qwxxxxxxxxxxx-zzzzzzz
```

But since I only want to setup one stream for one peer, thus the connection id is used instead of stream ID.